### PR TITLE
Add --debug-show-memo and --challenge-start to chia plots check

### PR DESCRIPTION
**Pull request by eFishCent**
- Added `chia plots check --debug-show-memo` to display memo in INFO log to allow recreating the same exact plot for debugging purposes.
- Added `chia plots check --challenge-start [start]` that begins at a different `[start]` for `-n [challenges]`. Useful when you want to do more detailed checks on plots without restarting from lower challenge values you already have done.

### CODE REVIEW

File: src/plotting/check_plots.py [hunk 1]
```diff
@@ -75,7 +87,7 @@ def check_plots(args, root_path):
         log.info(f"\tLocal sk: {plot_info.local_sk}")
         total_proofs = 0
         try:
-            for i in range(num):
+            for i in range(num_start, num):
```
Comment (mariano54) [inline src/plotting/check_plots.py:90]: Wouldn't it make more sense to go from num_start to num_start + num?
If I do --challenge-start 1000 -num 10  I expect it to go from 1000 to 1010.

**Comment by mariano54** [comment]
This test seems flaky FAILED tests/wallet/cc_wallet/test_cc_wallet.py::TestCCWallet::test_cc_max_amount_send
@Yostra

File: src/plotting/check_plots.py (hunk 1 above)
Comment (eFishCent) [reply src/plotting/check_plots.py]: You are right. Changed it.

**Review by mariano54 — APPROVED** [review]
Thanks for your contribution

**Pull request merged**